File: leetcode/strings/1221. Split a String in Balanced Strings/python/main.py
class Solution:
    def balancedStringSplit(self, s: str) -> int: 
        ans = 0
        if s.count("R") == s.count("L"):
            balance = 0
            for c in s:
                if c == "R":
                    balance += 1
                else:
                    balance -= 1
                if balance == 0:
                    ans += 1
        return ans


s = "RLRRLLRLRL"
print(Solution().balancedStringSplit(s))

File: leetcode/strings/1221. Split a String in Balanced Strings/golang/main.go
package main

import (
	"fmt"
	"strings"
)

func balancedStringSplit(s string) int {
	ans := 0
	if strings.Count(s, "L") == strings.Count(s, "R") {
		balance := 0
		for _, c := range s {
			if string(c) == "R" {
				balance++
			} else {
				balance--
			}
			if balance == 0 {
				ans++
			}
		}
	}

	return ans
}
func main() {
	fmt.Println(balancedStringSplit("RLRRLLRLRL"))
}
